controllers/componentdetectionquery_controller.go

```diff
@@ -145,7 +145,7 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 			} else {
 				// Use SPI to retrieve the devfile from the private repository
 				// TODO - maysunfaisal also search for Dockerfile
-				devfileBytes, err = spi.DownloadDevfileUsingSPI(r.SPIClient, ctx, componentDetectionQuery.Namespace, source.URL, "main", context)
+				devfileBytes, err = spi.DownloadDevfileUsingSPI(r.SPIClient, ctx, componentDetectionQuery.Namespace, source.URL, source.Revision, context)
```

Comment:
I'm wondering if no `Revision` has been passed in, will this cause any error?
we might want to add more unit tests on `Revision =""`,  and `Revision=<other than main>`

Reply:
Yes good find. The problem is that the actual SPI client requires a running Kube cluster (with SPI installed) in order to function properly, so it's not something that we can really test in the controller tests.

Reply:
Updated.